src/ProductBundle/Controller/CollectionController.php

```diff
@@ -21,92 +25,105 @@
 class CollectionController extends Controller
 {
     /**
-     * List the main collections.
-     *
-     * @return Response
+     * @var SeoPage
      */
-    public function indexAction(Request $request)
-    {
-        $pager = $this->get('sonata.classification.manager.collection')
-            ->getRootCollectionsPager($request->get('page'));
+    private $sonataSeoPage;
 
-        return $this->render('@SonataProduct/Collection/index.html.twig', [
-            'pager' => $pager,
-        ]);
+    /**
+     * @var CurrencyDetectorInterface
+     */
+    private $currencyDetector;
+
+    /**
+     * @var CollectionManagerInterface
+     */
+    private $collectionManagerInterface;
+
+    /**
+     * @var ProductSetManager
+     */
+    private $productSetManager;
+
+    public function __construct(SeoPage $sonataSeoPage = null, CurrencyDetectorInterface $currencyDetector = null, CollectionManagerInterface $collectionManagerInterface = null, ProductSetManager $productSetManager = null)
+    {
+        $this->sonataSeoPage = $sonataSeoPage;
+        $this->currencyDetector = $currencyDetector;
+        $this->collectionManagerInterface = $collectionManagerInterface;
+        $this->productSetManager = $productSetManager;
     }
 
     /**
-     * Display one collection.
-     *
-     * @param $collectionId
-     * @param $slug
-     *
-     * @throws NotFoundHttpException
+     * List the product related to one collection.
      *
      * @return Response
      */
-    public function viewAction($collectionId, $slug)
+    public function indexAction(Request $request)
```

Comment:
Technically this is a BC break. Can you please create proxy methods, that are calling the new methods?